src/inmanta/references.py

```diff
@@ -301,14 +334,37 @@ def serialize(self) -> SerializedReferenceLike:
 
     def serialize_arguments(self) -> Tuple[uuid.UUID, list[ArgumentTypes]]:
         """Serialize the arguments to this class"""
+
+        # The handling of references here is a but subtle:
+        # We replace every reference with a ReferenceArgument that refers to the reference by id
+        # The reference itself is not handled here but in inmanta.resources.ReferenceSubCollector.collect_reference
+        # There, the raw, unserialized tree of arguments is iterated over as well to collect the references themselves
+        # The caches on the Reference prevent this from being too inefficient by serializing only once
```

Comment:
I need to let this sink in.

Reply:
Do we really need a dedicated argument with dedicated mutation? Can't we extend `ReplaceValue` to accept `Resource | JsonArgument` and add one of those to the (real) collector?

Reply:
Is this true? Isn't the point of ReplaceValue to support nested references? 
 -> it is to support references in resources, but not references in references  (i.e. the not nested part is true)